### PR TITLE
Update render function

### PR DESCRIPTION
**Pull request by frogtd**
By my benchmarks, this is 3.7x faster than the other render function.

### CODE REVIEW

**Comment by Legend-of-iPhoenix** [comment]
I fixed some of the compilation errors but this does not pass the tests:

```
running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s


running 1 test
test readme_example ... FAILED

failures:

---- readme_example stdout ----
thread 'readme_example' panicked at 'assertion failed: `(left == right)`
  left: `" is a movie about ."`,
 right: `"The Birds is a movie about a swarm of birds that suddenly and violently attack the residents of a California coastal town."`', tests/readme_example.rs:21:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    readme_example

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
```

**Comment by frogtd** [comment]
That is odd, I'm not at my computer right now so I can't see why. I copy and pasted all of your tests while doing it locally so I don't know why it randomly failed.

File: src/lib.rs [hunk 1]
```diff
@@ -1,40 +1,34 @@
...
 
-        let mut iter = input[first..].chars();
-        while let Some(c) = iter.next() {
-            match c {
-                '{' => output.push_str(resolve_substitution(&mut iter, &context)),
-                _ => output.push(c),
+            for (y_index, y) in &mut iter {
+                if y == b'}' {
+                    let field_str = context.get_field(unsafe {
+                        std::str::from_utf8_unchecked(string.as_bytes().get_unchecked(last_index..x_index))
```
Comment (Legend-of-iPhoenix) [inline src/lib.rs]: This line is where the bug is, the range should be `x_index+1..y_index`

Comment (frogtd) [reply src/lib.rs]: I see, I copy and pasted that from the open above and issues arose.

**Pull request merged**